stout/event-loop.h

```diff
@@ -132,6 +132,43 @@ class EventLoop : public Scheduler {
     uv_buf_t buffer_ = {};
   };
 
+  class Signal {
+   public:
+    Signal(EventLoop& loop) {
+      handle_ = new uv_signal_t();
+      CHECK_EQ(
+          uv_signal_init(loop, handle_),
+          0);
+    }
+
+    Signal(const Signal& that) = delete;
+
+    Signal(Signal&& that) {
+      handle_ = that.handle_;
+      that.handle_ = nullptr;
+    }
+
+    ~Signal() {
+      if (handle_) {
+        if (uv_is_active(reinterpret_cast<uv_handle_t*>(handle_))) {
+          uv_signal_stop(handle_);
+        }
+
+        uv_close(
+            reinterpret_cast<uv_handle_t*>(handle_),
+            EventLoop::CloseCallback);
+      }
+    }
+
+    operator uv_signal_t*() {
+      CHECK(handle_);
+      return handle_;
+    }
```

Comment:
Do we need this operator overload? If not, I prefer to avoid operator overloads generally.

We could replace it with a named `handle()` method?

Reply:
I would also like to avoid doing this:
`uv_handle_set_data(reinterpret_cast<uv_handle_t*>(data.signal.handle()), &data)`

The problem is that you can't just pass the underlying handle ptr to certain libuv functions without converting them to uv_handle_t*.

How about `handle()` and `base_handle()`?